src/main/java/org/threadly/litesockets/SocketExecuter.java

```diff
@@ -0,0 +1,143 @@
+import org.threadly.util.Service;
+
+
+/**
+ * Basic SocketExecuter interface.  Different Implementations have different ways of operating, but
+ * they must all implement this for clients/servers to operate against them.
+ * 
+ * @author lwahlmeier
+ *
+ */
+public interface SocketExecuter extends Service {
+  
+  /**
+   * This will create a UDPServer on the specified {@link SocketExecuter}.
+   * 
+   * Please note that UDPClients are made from UDPServers not directly on the {@link SocketExecuter} 
+   * 
+   * @param host The local host or IP the UDPServer should listen on.
+   * @param port The local port the UDPServer should listen on.
+   * @return A {@link UDPServer}.
+   * @throws IOException This is only thrown if we can not create the UDPServers socket.
+   */
+  public UDPServer createUDPServer(String host, int port) throws IOException;
+  
+  /**
+   * This will create a {@link TCPServer} on the specified {@link SocketExecuter}.
+   * 
+   * @param host The local host or IP the TCPServer should listen on.
+   * @param port The local port the TCPServer should listen on.
+   * @return a {@link TCPServer}.
+   * @throws IOException This is only thrown if we can not create the TCPServers socket.
+   */
+  public TCPServer createTCPServer(String host, int port) throws IOException;
+  
+  /**
+   * This will crate a {@link TCPServer} from a given {@link ServerSocketChannel} object.
+   * 
+   * @param ssc The {@link ServerSocketChannel} object to use for this TCPServer. 
+   * @return a {@link TCPServer}.
+   * @throws IOException This is thrown if there is a problem with the {@link ServerSocketChannel}. 
+   */
+  public TCPServer createTCPServer(ServerSocketChannel ssc) throws IOException;
+  
+  /**
+   * This will create a {@link TCPClient} on the specified {@link SocketExecuter}.
+   * 
+   * @param host This is the remote host or IP to connect this client too.
+   * @param port This is the remote port to connect this client too.
+   * @return a {@link TCPClient} object.
+   * @throws IOException This is thrown if we can not make the socket or can not connect to the remote host.
+   */
+  public TCPClient createTCPClient(String host, int port) throws IOException;
+  
+  /**
+   * This will create a {@link TCPClient} on the specified {@link SocketExecuter} with
+   * an already created {@link SocketChannel}.  This is generally how a TCPServer will create its TCPClients.
+   * 
+   * @param sc the {@link SocketChannel} to be used for this TCPClient.
+   * @return a {@link TCPClient}
+   * @throws IOException This is thrown if there is a problem with the passed in {@link SocketChannel}.
+   */
+  public TCPClient createTCPClient(SocketChannel sc) throws IOException;
+  
+  /**
+   * This allows you to get the {@link Executor} for a specified object.
+   * 
+   * @param obj The Object whose {@link Executor} you are looking for
+   * @return the {@link Executor} for that object.
+   */
+  public Executor getExecutorFor(Object obj);
```

Comment:
Are "client" objects the only thing passed in here?  It might make sense to make this interface a bit more specific.  For example we would not want someone to get confused "do I pass in the client object?  or the socket address, or?".  So if we can make it more specific than Object, I think that would make it easier to understand.